Tests/SonarQube.TeamBuild.PreProcessor.Tests/DataModel/QualityProfile.cs

```diff
@@ -23,14 +24,20 @@ public QualityProfile(string name, string language)
         {
             this.name = name;
             this.language = language;
-            this.projectKeys = new HashSet<string>();
+            this.projectIds = new HashSet<string>();
             this.activeRules = new HashSet<Rule>();
             this.formatToContentExportMap = new Dictionary<string, string>();
         }
 
-        public QualityProfile AddProject(string projectKey)
+        public QualityProfile AddProject(string projectKey, string projectBranch = null)
```

Comment:
Previously there was no way to be branch-specific in a quality profile assignment. Rather than go through the complex task of creating a data structure that supports the relationship, it is simpler (and, for the moment, just as effective) to concatenate the key:branch to make a projectId.